examples/rpc/rpc_from_vcn_module/main.tf

```
@@ -0,0 +1,143 @@
+  
+  # general oci parameters
+  compartment_id = var.compartment_id
+  label_prefix   = "local"
+  freeform_tags  = var.freeform_tags
+
+  # vcn parameters
+  create_drg               = true
+  create_rpc               = true           
+  create_internet_gateway  = false           
+  lockdown_default_seclist = false           
+  create_nat_gateway       = true           
+  create_service_gateway   = false           
+  vcn_cidrs                = tolist([var.local_vcn_cidr])
+  vcn_dns_label            = "vcnlocal"
+  vcn_name                 = "vcnlocal"
+  nat_gateway_route_rules = [ 
+    {
+      destination       = var.remote_vcn_cidr # set remote vcn cidr as destination cidr 
+      destination_type  = "CIDR_BLOCK"     
+      network_entity_id = "drg"            
+      description       = "Terraformed - User added Routing Rule to remote vcn through DRG"
+    },
+  ]
+  providers = {
+    oci = oci
+  }
+  
+}
+
+module "subnet_local" {
+  source          = "github.com/oracle-terraform-modules/terraform-oci-tdf-subnet"
+  
+  default_compartment_id  = var.compartment_id
+  vcn_id                  = module.vcn_local.vcn_id
+  vcn_cidr                = var.local_vcn_cidr
+  
+  subnets = {
+    local_rpc_subnet               = {
+      compartment_id    = null
+      defined_tags      = null
+      freeform_tags     = null
+      dynamic_cidr      = false
+      cidr              = var.local_vcn_cidr
+      cidr_len          = null
+      cidr_num          = null
+      enable_dns        = true
+      dns_label         = "localrpcsubnet"
+      private           = true
+      ad                = null
+      dhcp_options_id   = null
+      route_table_id    = module.vcn_local.nat_route_id
+      security_list_ids = null
+    },
+  }
+}
+
+
+module "vcn_remote" {
+  # this module use the generic vcn module and configure it to act as remote vcn
+  source  = "../../../"
+  
+  # general oci parameters
+  compartment_id = var.compartment_id
+  label_prefix   = "remote"
+  freeform_tags  = var.freeform_tags
+
+  # vcn parameters
+  create_drg               = true           
+  create_rpc               = true
+  create_internet_gateway  = true           
+  lockdown_default_seclist = false           
+  create_nat_gateway       = false           
+  create_service_gateway   = false           
+  vcn_cidrs                = tolist([var.remote_vcn_cidr]) # VCN CIDR
+  vcn_dns_label            = "vcnremote"
+  vcn_name                 = "vcnremote"
+  internet_gateway_route_rules = [ 
+    {
+      destination       = var.local_vcn_cidr # set local vcn cidr as destination cidr 
+      destination_type  = "CIDR_BLOCK"     
+      network_entity_id = "drg"            
+      description       = "Terraformed - User added Routing Rule to local vcn through DRG"
+    },
+  ]
+  drg_rpc_id_remote = module.vcn_local.rpc_id
+  drg_rpc_region_remote = var.region
+
+  providers = {
+    oci = oci.remote
+  }
+  
+}
+
+module "subnet_remote" {
```

Please use resources instead of external modules we don't maintain directly. This is to reduce external dependencies.
```suggestion
resource "oci_core_subnet" "remote" {
```